src/test/java/edu/harvard/iq/dataverse/engine/command/impl/RestrictFileCommandTest.java

```diff
@@ -89,19 +91,33 @@ public void testRestrictUnpublishedFile() throws CommandException{
     
     @Test
     public void testRestrictPublishedFile() throws Exception{
-        file.setOwner(dataset);
         dataset.setPublicationDate(new Timestamp(new Date().getTime()));
+        // Restrict on a published file will cause the creation of a new draft dataset version
+        // and should update only the FileMetadata in the draft version for the test file.
+        // So we need to make sure that we use one of the files in the dataset for the test 
+        DataFile file = dataset.getFiles().get(0);
+        // And make sure is is file.isReleased() == true
+        file.setPublicationDate(dataset.getPublicationDate());
+        // And set its owner, which is usually done automatically, but not in the test setup
+        file.setOwner(dataset);
+        //And set the version state to released so that the RestrictFileCommand will create a draft version
+        dataset.getLatestVersion().setVersionState(VersionState.RELEASED);
         RestrictFileCommand cmd = new RestrictFileCommand(file, makeRequest(), restrict);
         engine.submit(cmd);
 
         //asserts
         assertTrue(!file.isRestricted());
+        boolean fileFound = false;
         for (FileMetadata fmw : dataset.getEditVersion().getFileMetadatas()) {
             if (file.equals(fmw.getDataFile())) {
-                assertEquals(fmw, file.getFileMetadata());
+                fileFound=true;
+                //If it worked fmw is for the draft version and file.getFileMetadata() is for the published version
                 assertTrue(fmw.isRestricted());
+                assertTrue(!file.getFileMetadata().isRestricted());
+                break;
             }
         }
+        assertTrue(fileFound);
```

Comment:
The updated asserts make sure that the file is found in the dataset and verify that only the file metadata for the new draft version is updated to restricted.